src/System.Memory/src/System/Buffers/SequenceReader.cs

```diff
@@ -24,7 +23,7 @@ public SequenceReader(ReadOnlySequence<T> sequence)
             Consumed = 0;
             Sequence = sequence;
             _currentPosition = sequence.Start;
-            _length = -1;
+            Length = Sequence.Length;
```

Comment:
Are we fine with making the ctor more expensive here?

Reply:
I can reintroduce the caching with unsafe I believe? I'll double check, but otherwise, yes.

Reply:
@tannergooding you mentioned something about using unsafe in this context? I can't seem to figure out how to get that to compile.

Reply:
You just use unsafe code to lazily initialize the value in the getter such that it is still lazily initialized but is functionally readonly as far as the user can see.

See https://github.com/dotnet/corefx/pull/37737#discussion_r285243100

Reply:
Thanks! Should I be using `Volatile.Write` in this case (`long`)?

Reply:
I think @jkotas just used `Volatile.Write` since `MethodInfo` is a reference type and that you should be fine using `Unsafe.Write` or a regular pointer; but he might also have had a good reason for using `Volatile` instead and may be able to comment...

Reply:
I have used `Volatile.Write` in the other change because the field was volatile before my change and because it shows the intent better.